pkg/runtime/processor/processor_test.go

```diff
@@ -462,3 +487,170 @@ func TestProcessNoWorkflow(t *testing.T) {
 	_, ok := proc.managers[components.CategoryWorkflow]
 	require.False(t, ok, "workflow cannot be registered as user facing component")
 }
+
+func TestReporter(t *testing.T) {
+	pubsubComponent := componentsapi.Component{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testpubsub",
+		},
+		Spec: componentsapi.ComponentSpec{
+			Type:        "pubsub.mockPubSub",
+			Version:     "v1",
+			Metadata:    daprt.GetFakeMetadataItems(),
+			InitTimeout: "2",
+		},
+	}
+
+	t.Run("event is reported on successful Init",
+		func(t *testing.T) {
+			resultChan := make(chan operatorv1.ResourceResult, 1)
+			proc, reg := newTestProc(
+				withReporter(func(_ context.Context, _ componentsapi.Component, result operatorv1.ResourceResult) error {
+					resultChan <- result
+					return nil
+				}))
+
+			mockPubSub := new(daprt.MockPubSub)
+			reg.PubSubs().RegisterComponent(
+				func(_ logger.Logger) pubsub.PubSub {
+					return mockPubSub
+				},
+				"mockPubSub",
+			)
+
+			mockPubSub.On("Init", mock.Anything).Return(nil)
+			mockPubSub.On("Close").Return(nil)
+
+			err := proc.processComponentAndDependents(context.Background(), pubsubComponent)
+			require.NoError(t, err)
+
+			select {
+			case result := <-resultChan:
+				assert.Equal(t, operatorv1.ResourceType_COMPONENT, result.ResourceType)
+				assert.Equal(t, operatorv1.EventType_INIT, result.EventType)
+				assert.Equal(t, operatorv1.ResourceConditionStatus_SUCCESS, result.Condition)
+				assert.Equal(t, pubsubComponent.Name, result.Name)
+			case <-time.After(20 * time.Second):
```

Comment:
Hopefully computers are quicker than this.
```suggestion
			case <-time.After(5 * time.Second):
```